src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -160,6 +161,42 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             if (otlpHttpEndpointUrl != null)
             {
                 context.EnvironmentVariables[DashboardConfigNames.DashboardOtlpHttpUrlName.EnvVarName] = otlpHttpEndpointUrl;
+
+                var model = context.ExecutionContext.ServiceProvider.GetRequiredService<DistributedApplicationModel>();
+                var allResourceEndpoints = model.Resources
+                    .Where(r => !string.Equals(r.Name, KnownResourceNames.AspireDashboard, StringComparisons.ResourceName))
+                    .SelectMany(r => r.Annotations)
+                    .OfType<EndpointAnnotation>()
+                    .ToList();
+
+                var corsOrigins = new HashSet<string>();
+                foreach (var endpoint in allResourceEndpoints)
+                {
+                    if (endpoint.UriScheme is "http" or "https")
+                    {
+                        // Prefer allocated endpoint over EndpointAnnotation.Port.
+                        var origin = endpoint.AllocatedEndpoint?.UriString;
+                        if (origin == null && endpoint.Port != null)
+                        {
+                            origin = $"{endpoint.UriScheme}://localhost:{endpoint.Port}";
```

Comment:
`localhost` could have an extreme corner case here. Lets say someone spins up a headless browser in a container (perhaps for test automation). Then localhost will likely be the wrong hostname (platform dependent).

Reply:
Let's wait and see whether there is feedback about the automatic CORS policy not working. Can figure out improvements then.